src/core/room.ts

```diff
@@ -178,13 +183,26 @@ export class DefaultRoom implements Room {
 
     if (options.presence) {
       this._logger.debug('enabling presence on room', { roomId });
-      this._presence = new DefaultPresence(roomId, channelManager, realtime.auth.clientId, logger);
+      this._presence = new DefaultPresence(
+        roomId,
+        channelManager,
+        defaultPresenceManager,
+        realtime.auth.clientId,
+        logger,
+      );
       features.push(this._presence);
     }
 
-    if (options.typing) {
+    if (options.presence?.typingOptions) {
       this._logger.debug('enabling typing on room', { roomId });
-      this._typing = new DefaultTyping(roomId, options.typing, channelManager, realtime.auth.clientId, logger);
+      this._typing = new DefaultTyping(
```

Comment:
If we're going to conceptualise this as being a part of presence, then it should be structured as `room.presence.typing`, or just having methods directly on `room.presence`